compiler-plugin-tests/src/test/java/io/ballerina/stdlib/http/compiler/CompilerPluginTest.java

```diff
@@ -774,4 +774,12 @@ public void testPathParameterType() {
                 " 'string','int','float','decimal','boolean' types or a rest parameter with one of the above types",
                 HTTP_145);
     }
+
+    @Test
+    public void testResourceSignatureParamValidations() {
+        Package currentPackage = loadPackage("sample_package_38");
+        PackageCompilation compilation = currentPackage.getCompilation();
+        DiagnosticResult diagnosticResult = compilation.diagnosticResult();
+        Assert.assertEquals(diagnosticResult.errorCount(), 36);
```

Comment:
dont we need to validate the error messages?

Reply:
Will add the error message assertions 👍🏼

Reply:
Fixed via [bef7551](https://github.com/ballerina-platform/module-ballerina-http/pull/1603/commits/bef7551ee5309fd972cc7846e72f1ed3e3f56713)